Realm/RLMApp.h

```diff
@@ -0,0 +1,97 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMNetworkTransporting.h"
+
+#ifndef RLMApp_h
+#define RLMApp_h
+
+@class RLMSyncUser, RLMAppCredentials;
+
+/// A block type used for APIs which asynchronously vend an `RLMSyncUser`.
+typedef void(^RLMUserCompletionBlock)(RLMSyncUser * _Nullable, NSError * _Nullable);
+
+#pragma mark RLMAppConfiguration
+
+/// Properties representing the configuration of a client
+/// that communicate with a particular Realm application.
+@interface RLMAppConfiguration : NSObject
+
+/// A custom base URL to request against.
+@property (nonatomic, strong) NSString* _Nullable baseURL;
+
+/// A transport for customizing network handling.
+@property (nonatomic, strong) id <RLMNetworkTransporting> _Nullable transport;
+
+/// A custom app name.
+@property (nonatomic, strong) NSString* _Nullable localAppName;
+
+/// A custom app version.
+@property (nonatomic, strong) NSString* _Nullable localAppVersion;
+
+/// The default timeout for network requests.
+@property (nonatomic, assign) NSUInteger defaultRequestTimeoutMS;
+
+-(instancetype _Nonnull) initWithBaseURL:(NSString* _Nullable) baseURL
```

Comment:
Each header file should normally be wrapped in `NS_ASSUME_NONNULL_BEGIN`/`NS_ASSUME_NONNULL_END` so that things are non-null by default rather than marking individual things as nonnull.

Space between - and (, no space between ) and argument name.